include/orbis/_types/net.h

```diff
@@ -1,11 +1,108 @@
+#define ORBIS_NET_IP_MAXTTL             31
+#define ORBIS_NET_IP_DONTFRAG           67
+
+/* socket option (TCP) */
+#define ORBIS_NET_TCP_NODELAY           1
+#define ORBIS_NET_TCP_MAXSEG            2
+#define ORBIS_NET_TCP_MSS_TO_ADVERTISE  3
+#define ORBIS_NET_TCP_KEEPLISTEN        128
+
+/* socket option (UDP) */
+#define ORBIS_NET_UDP_SND_ON_SUSPEND    128
+
+/* socket option (SOCKET) */
+#define ORBIS_NET_SO_REUSEADDR          0x00000004
+#define ORBIS_NET_SO_KEEPALIVE          0x00000008
+#define ORBIS_NET_SO_BROADCAST          0x00000020
+#define ORBIS_NET_SO_LINGER             0x00000080
+#define ORBIS_NET_SO_REUSEPORT          0x00000200
+#define ORBIS_NET_SO_ONESBCAST          0x00010000
+#define ORBIS_NET_SO_USECRYPTO          0x00020000
+#define ORBIS_NET_SO_USESIGNATURE       0x00040000
+#define	ORBIS_NET_SO_NOSIGPIPE			0x00000800
+
+#define ORBIS_NET_SO_SNDBUF             0x1001
+#define ORBIS_NET_SO_RCVBUF             0x1002
+#define ORBIS_NET_SO_ERROR              0x1007
+#define ORBIS_NET_SO_TYPE               0x1008
+
+#define ORBIS_NET_SO_SNDTIMEO           0x1105
+#define ORBIS_NET_SO_RCVTIMEO           0x1106
+#define ORBIS_NET_SO_ERROR_EX           0x1107
+#define ORBIS_NET_SO_ACCEPTTIMEO        0x1108
+#define ORBIS_NET_SO_CONNECTTIMEO       0x1109
+
+#define ORBIS_NET_SO_NBIO               0x1200
+#define ORBIS_NET_SO_POLICY             0x1201
+#define ORBIS_NET_SO_NAME               0x1202
+#define ORBIS_NET_SO_PRIORITY           0x1203
+
+/* socket types */
 #define ORBIS_NET_AF_INET				2
 #define ORBIS_NET_SOCK_STREAM			1
 #define ORBIS_NET_SOCK_DGRAM			2
 #define ORBIS_NET_SOCK_RAW				3
 #define ORBIS_NET_SOCK_DGRAM_P2P		6
 #define ORBIS_NET_SOCK_STREAM_P2P		10
+
+/* address families */
+#define ORBIS_NET_AF_LOCAL				1
+#define ORBIS_NET_AF_INET               2
+
+/* ORBIS_NET_SO_POLICY */
+#define ORBIS_NET_SOCK_POLICY_NUM_MIN   0
+#define ORBIS_NET_SOCK_POLICY_NUM_MAX   15
+#define ORBIS_NET_SOCK_POLICY_NA        -1
+
+/* ORBIS_NET_SO_PRIORITY */
+#define ORBIS_NET_SOCK_PRIORITY_NUM_MIN        1
+#define ORBIS_NET_SOCK_PRIORITY_NUM_USER_MIN   8
+#define ORBIS_NET_SOCK_PRIORITY_NUM_USER_MAX   23
+#define ORBIS_NET_SOCK_PRIORITY_NUM_MAX        31
+
+/* address & mask */
+#define ORBIS_NET_INADDR_ANY           ((unsigned int)0x00000000)
+#define ORBIS_NET_INADDR_LOOPBACK      ((unsigned int)0x7f000001)
+#define ORBIS_NET_INADDR_BROADCAST     ((unsigned int)0xffffffff)
+#define ORBIS_NET_INADDR_UNSPEC_GROUP  ((unsigned int)0xe0000000)
+#define ORBIS_NET_INADDR_AUTOIP        ((unsigned int)0xa9fe0000)
+
+#define ORBIS_NET_IN_CLASSD_NET        ((unsigned int)0xf0000000)
+#define ORBIS_NET_IN_AUTOIP_NET        ((unsigned int)0xffff0000)
+
+#define ORBIS_NET_INET_ADDRSTRLEN     16
+
+/* multicast */
+#define ORBIS_NET_IP_DEFAULT_MULTICAST_TTL   1
+#define ORBIS_NET_IP_DEFAULT_MULTICAST_LOOP  1
+
+/* flags */
+#define ORBIS_NET_MSG_PEEK            0x00000002
+#define ORBIS_NET_MSG_WAITALL         0x00000040
```

Comment:
if @OSM-Made values are right, then we need to update `MSG_xxx` definitions under socket.h. For example,
```
/include/sys/socket.h:306:#define MSG_WAITALL   0x0100
```

some other discrepancies (I didn't check everything, but I'm sure other values will need fix too), like:
```c
#define SO_RCVTIMEO     0x1006
#define SO_SNDTIMEO     0x1005
```